test/test_automl/test_automl.py

```diff
@@ -156,7 +159,7 @@ def test_delete_non_winning_models(self):
         self.assertTrue(ensemble_members_idx.issubset(model_files_idx))
 
         del automl
-        self._tearDown(backend_api.temporary_directory)
+        #self._tearDown(backend_api.temporary_directory)
```

Comment:
Is there a reason to comment this line?

Reply:
Sorry man. This last commit is a work in progress. I should have told you. (I tried to put the PR back to "draft", but couldn't)
So maybe don't bother reviewing it right now. It still has bugs.
The only reason this line is commented out is for debugging.

Reply:
No worries, I should have waited for your review request ;)

Reply:
Hey, I just realized that with the version which is currently in the development branch, the ensemble builder script can sometimes crash due to a file no longer existing. This happens ~line 450, in the following code:
```python
        for k, _, _ in sorted_keys[:keep_nbest]:
            if self.read_preds[k][Y_ENSEMBLE] is None:
                self.read_preds[k][Y_ENSEMBLE] = self._read_np_fn(fp=k)
                # No need to load valid and test here because they are loaded
                #  only if the model ends up in the ensemble
            self.read_preds[k]['loaded'] = 1
```
I put around a try/except:
```python
        try:
            for k, _, _ in sorted_keys[:keep_nbest]:
                if self.read_preds[k][Y_ENSEMBLE] is None:
                    self.read_preds[k][Y_ENSEMBLE] = self._read_np_fn(fp=k)
                    # No need to load valid and test here because they are loaded
                    #  only if the model ends up in the ensemble
                self.read_preds[k]['loaded'] = 1
        except Exception as e:
            print('*' * 80)
            print('*' * 80)
            print('*' * 80)
            print('*' * 80)
            print('*' * 80)
            print(e)
            raise e
```
could you please check in your version that this isn't triggered?

Reply:
So far I haven't noticed anything. But I'll investigate.

Reply:
Right now I'm working on a bug that I have no clue why it is happening. But it is caused by the locking logic that I added to the `AbstractEvaluator`.